src/react-infinite.jsx

```diff
@@ -287,9 +288,10 @@ var Infinite = React.createClass({
   },
 
   componentDidUpdate(prevProps: ReactInfiniteProps, prevState: ReactInfiniteState) {
-    this.loadingSpinnerHeight = this.utils.getLoadingSpinnerHeight();
-
     if (this.props.displayBottomUpwards) {
+      // this line was in the top of this function, moved to here for performance improvments
+      this.loadingSpinnerHeight = this.utils.getLoadingSpinnerHeight();
```

Comment:
They only used this property in all the code when displayBottomUpwards === true, so i think that there were unnecessary logic triggered.